src/lib.rs

```diff
@@ -3632,6 +3907,31 @@ impl Connection {
                 // Once the handshake is confirmed, we can drop Handshake keys.
                 self.drop_epoch_state(packet::EPOCH_HANDSHAKE, now);
             },
+
+            frame::Frame::Datagram { data } => {
+                // Close the connection if DATAGRAMs are not enabled.
+                // quiche always advertises support for 64K sized DATAGRAM
+                // frames, as recommended by the standard, so we don't need a
+                // size check.
+                if self
+                    .local_transport_params
+                    .max_datagram_frame_size
+                    .is_none()
```

Comment:
Not sure I get it, this specific `if` is to address this part of the draft:

> An endpoint that receives a DATAGRAM frame when it has not sent the max_datagram_frame_size transport parameter MUST terminate the connection with error PROTOCOL_VIOLATION.

That's the reason this case exits with `Error::InvalidState`. The full queue state ends up with an `Ok(())` and the head of the queue be discarded.

Reply:
I don't understand the original comment about queues. This is about detecting if the application enabled datagram reception using config/TPs as noted.

I actually hit this while porting the apps into quiche-apps :)

Reply:
So, what I meant basically was that we could check if the queue capacity is `0` instead of the transport param, because I didn't like how this looks :) (the `is_full()` part was just because I got confused on how it works).

But I guess it's the correct check to do, so I'm fine leaving as-is.

Reply:
How about I add a `datagram_enabled()` method that can be used for the check here, then internally it can test TPs or queues as we decide later on?

Reply:
I've done this